server/lib/SignatureHandler.php

```diff
@@ -97,4 +98,33 @@ private function splitCloudId(string $cloudId): array {
 		return [$user, $host];
 	}
 
+
+	/**
+	 * @param Request $request
+	 *
+	 * @throws SignedRequestException
+	 */
+	public function verifyRequest(Request $request): string {
+		$body = json_decode($request->getBody(), true);
+		if ($body === null || !isset($body['message']) || !isset($body['message']['data'])
+			|| !isset($body['message']['data']['federationId'])
+			|| !isset($body['signature'])
+			|| !isset($body['message']['timestamp'])) {
```

Comment:
```suggestion
		if ($body === null ||
			|| !isset($body['message']['data']['federationId'])
			|| !isset($body['signature'])
			|| !isset($body['message']['timestamp'])) {
```
isset does not need to test each intermediate array